### PR TITLE
react-validated-input-component - Roland Estropia Jr.

### PR DESCRIPTION
**Pull request by restropiajr**
Recording 2023-06-27 at 20 33 45

### CODE REVIEW

File: react-validated-input-component/my-app/src/ValidatedInput.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { ChangeEvent, useState, KeyboardEvent } from 'react';
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState({
+    message: '',
+    displayStyle: '',
+    color: '',
+  });
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.tsx]: In general, I suggest moving stuff out of state that can be directly calculated based on other state / props, to reduce the chance of introducing state inconsistencies (which are difficult to debug). For example, the value of `displayStyle` can be calculated directly based on `password` / `message`.

File: react-validated-input-component/my-app/src/ValidatedInput.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        };
+      }
+    });
+  }
+
+  function handleKeyDown(event: KeyboardEvent<HTMLInputElement>) {
+    if (event.key === 'Enter' && event.currentTarget.type === 'password') {
+      event.preventDefault();
+    }
+  }
```
Comment (TimelikeClosure) [inline react-validated-input-component/my-app/src/ValidatedInput.tsx]: I'd suggest tying this to the form's `'submit'` event whenever there's a form wrapping an input.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: react-validated-input-component/my-app/src/ValidatedInput.tsx (hunk 1 above)
Comment (restropiajr) [reply react-validated-input-component/my-app/src/ValidatedInput.tsx]: @TimelikeClosure updated

File: react-validated-input-component/my-app/src/ValidatedInput.tsx (hunk 2 above)
Comment (restropiajr) [reply react-validated-input-component/my-app/src/ValidatedInput.tsx]: @TimelikeClosure updated

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**